src/Symfony/Component/Mailer/Mailer.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer;
+
+use Symfony\Component\Mailer\Messenger\EnvelopedMessage;
+use Symfony\Component\Mailer\Transport\TransportInterface;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Mime\RawMessage;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @experimental in 4.3
+ */
+class Mailer implements MailerInterface
+{
+    private $transport;
+    private $bus;
+
+    public function __construct(TransportInterface $transport, MessageBusInterface $bus = null)
+    {
+        $this->transport = $transport;
+        $this->bus = $bus;
+    }
+
+    public function send(RawMessage $message, SmtpEnvelope $envelope = null): void
+    {
+        if (null === $this->bus) {
+            $this->transport->send($message, $envelope);
+
+            return;
+        }
+
+        $this->bus->dispatch(new EnvelopedMessage($message, $envelope));
```

Comment:
is `EnvelopedMessage` internal? Should it? Or am i allowed to do dispatch it without using a mailer .. if so i'd rename it to `SendEmailMessage()`

related to above comment; i tend to believe one uses either a mailer or a bus as entry point.

Reply:
You can dispatch yourself if you want to. Renamed.